### PR TITLE
Add paginated feed of event streams

### PR DESCRIPTION
**Pull request by jaceko**

### CODE REVIEW

File: persistence/persistence-jdbc/src/main/java/uk/gov/justice/services/jdbc/persistence/PaginationCapableRepository.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package uk.gov.justice.services.jdbc.persistence;
+
+import java.util.stream.Stream;
+
+public interface PaginationCapableRepository<E> {
+    Stream<E> getPage(long offset, long pageSize);
```
Comment (matt-rich) [inline persistence/persistence-jdbc/src/main/java/uk/gov/justice/services/jdbc/persistence/PaginationCapableRepository.java]: final params?

File: event-sourcing/event-repository/event-repository-jdbc/src/main/java/uk/gov/justice/services/eventsourcing/repository/jdbc/eventstream/EventStreamJdbcRepository.java [hunk 2]
```diff
@@ -15,9 +16,10 @@
 import javax.naming.NamingException;
 
 @ApplicationScoped
-public class EventStreamJdbcRepository extends AbstractJdbcRepository<EventStream> {
+public class EventStreamJdbcRepository extends AbstractJdbcRepository<EventStream> implements PaginationCapableRepository<EventStream> {
     private static final String SQL_INSERT_EVENT_STREAM = "INSERT INTO event_stream (stream_id) values (?);";
     private static final String SQL_FIND_ALL = "SELECT * FROM event_stream ORDER BY sequence_number ASC;";
+    private static final String SQL_PAGE = "SELECT * FROM event_stream ORDER BY sequence_number ASC LIMIT ? OFFSET ?;";
 
     private static final String READING_STREAM_ALL_EXCEPTION = "Exception while reading stream";
```
Comment (jaceko) [inline event-sourcing/event-repository/event-repository-jdbc/src/main/java/uk/gov/justice/services/eventsourcing/repository/jdbc/eventstream/EventStreamJdbcRepository.java]: done

File: event-sourcing/event-source-api-integration-test/src/test/java/uk/gov/justice/services/eventsourcing/source/api/EventStreamsFeedIT.java [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+
+        assertThat(firstPageAgain.getStatusLine().getStatusCode(), is(200));
+
+        with(responseBodyOf(firstPageAgain))
+                .assertThat("$.data", hasSize(2))
+                .assertThat("$.data[0].streamId", is(streamId1.toString()))
+                .assertThat("$.data[1].streamId", is(streamId2.toString()));
+
+
+    }
```
Comment (jaceko) [inline event-sourcing/event-source-api-integration-test/src/test/java/uk/gov/justice/services/eventsourcing/source/api/EventStreamsFeedIT.java]: we've got unit tests for that in EventStreamsFeedServiceTest:
shouldNotReturnLinkToNextPageIfNoMoreRecords
shouldNotReturnLinkToPreviousPageIfOnFirstPage

File: event-sourcing/event-source-api/src/main/java/uk/gov/justice/services/eventsourcing/source/api/feed/eventstream/EventStreamEntry.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package uk.gov.justice.services.eventsourcing.source.api.feed.eventstream;
+
+import java.util.UUID;
+
+public class EventStreamEntry {
+    private final String streamId;
+    private final String href;
+
+    public EventStreamEntry(final UUID streamId, final String href) {
```
Comment (jaceko) [inline event-sourcing/event-source-api/src/main/java/uk/gov/justice/services/eventsourcing/source/api/feed/eventstream/EventStreamEntry.java]: This is a dto object to be serialised into json. Nothing to do with db.
I just want to make sure that the UUID gets serialised to json properly.

File: event-sourcing/event-source-api/src/main/raml/event-source-api.raml [hunk 5]
```diff
@@ -15,10 +15,10 @@ protocols: [ HTTP, HTTPS ]
             name: event-source.get-event-streams
         ...
     queryParameters:
-        ref:
```
Comment (jaceko) [inline event-sourcing/event-source-api/src/main/raml/event-source-api.raml]: done

**Pull request merged**